### PR TITLE
Implemented "nice" authentication in one button. Added endpoint comments

### PR DESCRIPTION
**Pull request by ErikasSilobritas**

### CODE REVIEW

File: DownloadMoreRAM/src/API/Controllers/CacheController.cs [hunk 1]
```diff
@@ -7,28 +7,37 @@ namespace API.Controllers;
 [Route("[controller]")]
 public class CacheController : ControllerBase
 {
+    ///<summary>
+    ///Create a dictionary
```
Comment (rusvaplauke) [inline DownloadMoreRAM/src/API/Controllers/CacheController.cs:11]: A more accurate phrase would be "add a dictionary entry" or similar.

**Review by rusvaplauke — CHANGES_REQUESTED** [review]
Some changes in comments

**Comment by rusvaplauke** [comment]
If I enter an invalid key, the pop-up still says "authorized" (even though later the API calls return an "invalid api key"). Is this expected behavior?

File: DownloadMoreRAM/src/Application/Services/PeriodicHostedService.cs
Comment (indregie) [inline DownloadMoreRAM/src/Application/Services/PeriodicHostedService.cs:1]: Undo these changes because there will be merge conflicts

File: DownloadMoreRAM/src/API/Program.cs [hunk 2]
```diff
@@ -20,6 +20,26 @@
...
+        In = ParameterLocation.Header,
+        Name = "X-Api-Key",
+        Type = SecuritySchemeType.ApiKey,
+        Description = "API key needed to access the endpoints."
+    });
+
+    // Add security requirement
+    c.AddSecurityRequirement(new OpenApiSecurityRequirement
+    {
+        {
```
[the remark this answers is not in the document]
Comment (ErikasSilobritas) [reply DownloadMoreRAM/src/API/Program.cs:35]: nope, they're needed here ;)

File: DownloadMoreRAM/src/Application/Services/PeriodicHostedService.cs
Comment (ErikasSilobritas) [reply DownloadMoreRAM/src/Application/Services/PeriodicHostedService.cs:1]: done

File: DownloadMoreRAM/src/API/Controllers/CacheController.cs (hunk 1 above)
Comment (ErikasSilobritas) [reply DownloadMoreRAM/src/API/Controllers/CacheController.cs:11]: done

File: DownloadMoreRAM/src/API/Controllers/CacheController.cs [hunk 3]
```diff
@@ -7,28 +7,37 @@ namespace API.Controllers;
...
-    public async Task<IActionResult> Get([FromHeader(Name = "X-Api-Key")] string userApiKey, string key)
+    public async Task<IActionResult> Get(string key)
     {
         return Ok("in progress");
 
         throw new NotImplementedException();
     }
-
+    ///<summary>
+    ///Delete a dictionary by key
```
[the remark this answers is not in the document]
Comment (ErikasSilobritas) [reply DownloadMoreRAM/src/API/Controllers/CacheController.cs:37]: ok

File: DownloadMoreRAM/src/API/Program.cs [hunk 4]
```diff
@@ -20,6 +20,26 @@
...
+
+    c.AddSecurityDefinition("ApiKey", new OpenApiSecurityScheme
+    {
+        In = ParameterLocation.Header,
+        Name = "X-Api-Key",
+        Type = SecuritySchemeType.ApiKey,
+        Description = "API key needed to access the endpoints."
+    });
+
+    // Add security requirement
```
[the remark this answers is not in the document]
Comment (ErikasSilobritas) [reply DownloadMoreRAM/src/API/Program.cs:32]: deleted
